src/nvim/screen.c

```diff
@@ -1615,8 +1615,8 @@ static void win_draw_end(win_T *wp, int c1, int c2, bool draw_margin, int row,
     }
   }
 
-  int attr = hl_combine_attr(wp->w_hl_attr_normal,
-                             hl ? win_hl_attr(wp, hl) : 0);
+  int hl_prim_attr =  hl != HLF_8 ? win_hl_attr(wp, hl) : 0;
+  int attr = hl_combine_attr(wp->w_hl_attr_normal, hl_prim_attr);
```

Comment:
I think @bfredl is suggesting:

```suggestion
  int attr = hl_combine_attr(wp->w_hl_attr_normal,
                             hl != -1 ? win_hl_attr(wp, hl) : 0);
```

Reply:
@bfredl @justinmk 
I am struggling to understand the semantic value of `hl != -1`. The enum starts at `0`, so this condition will always be true and we will get another PVS warning. 

Same for `HLF_COUNT` - condition will always be true.

Can we simply do like this?:

```c
int attr = hl_combine_attr(wp->w_hl_attr_normal, win_hl_attr(wp, hl))?

```

Reply:
Changing the sentinel value implies changing both this check and the places it is currently set (explicit zero).

Reply:
since `win_hl_attr` is accessing array by `hl` index it should be checked both for being >=0 and < `HLF_COUNT`

Reply:
It only uses hardcoded internal values. One sentinel value is enough.